samples/TagHelperSample.Web/TagHelpers/ConditionalCommentTagHelper.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+
+namespace TagHelperSample.Web
+{
+    [TargetElement("iecondition")]
+    public class ConditionalCommentTagHelper : TagHelper
+    {
+        [HtmlAttributeName("mode")]
+        public string Mode { get; set; }
```

Comment:
Talked in person: Conditional comments do different things based on the IE version. Change Mode into an enum as noted in my initial comment on this line.